### PR TITLE
fix special string in the filename

### PR DESCRIPTION
**Pull request by d-d-up**
Signed-off-by: d-d-up <qhr6113@163.com>

**What this PR does / why we need it**:
Fixes:issue https://github.com/helm/helm/issues/11196
**Special notes for your reviewer**:

**If applicable**:
- [ ] this PR contains documentation
- [ ] this PR contains unit tests
- [ ] this PR has been tested for backwards compatibility

### CODE REVIEW

**Review by mattfarina — APPROVED** [review]
Also tested manually with before and after. It does fix the panic.

**Comment by mattfarina** [comment]
Note, this needs an approval from a second helm maintainer before it can be merged.

**Review by hickeyma — APPROVED** [review]
Tested and it fixes the panic and provides relevant error message. Thanks for the fix @d-d-up

**Pull request merged**
